Jenkinsfile-testing

```diff
@@ -1,8 +1,15 @@
 def versionTag = "v${env.BUILD_NUMBER}"
 
+def setBuildStatus(String status, String desc) {
+  withCredentials([string(credentialsId: 'github-jenkins-access-token', variable: 'GITHUB_ACCESS_TOKEN')]) {
+    String sha = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+    String url = "https://api.github.com/repos/nationalarchives/tdr-transfer-frontend/statuses/${sha}"
+    sh "curl -XPOST '${url}' -H 'Authorization: bearer ${env.GITHUB_ACCESS_TOKEN}' --data '{\"state\":\"${status}\",\"target_url\":\"${env.BUILD_URL}\",\"description\":\"${desc}\",\"context\":\"Jenkins TDR Front end test\"}'"
```

Comment:
Did you look into whether any of the git plugins would do this step for us? The looks like it will work - I'm just wondering if Jenkins has a built-in tool.

Reply:
I did yeah, the plugins that I could find usually required a username/password rather than SSH/access keys or they required all GitHub users be 'whitelisted' to enable them to kick off a request (Which I thought didn't quite meet what we wanted). A few plugins also didn't work for multi-branch pipelines. I put together some notes of what I found and what they need if you'd like to look at them (Though I didn't take note of any that require a different Jenkins structure since I didn't feel it was necessary)?

Reply:
If you have some notes, can you stick them on the Jira card? That might be the best place for them if we ever need to revisit this.

Reply:
Will do! 😄